src/views/SystemProfile/components/ATO/index.tsx

```diff
@@ -45,22 +45,24 @@ import './index.scss';
 /**
  * Get counts of Security Findings from Cedar threat levels.
  */
-// eslint-disable-next-line camelcase
-function getSecurityFindings(cedarThreat: GetSystemProfile_cedarThreat[]) {
-  return cedarThreat.reduce<SecurityFindings>(
-    (prev: SecurityFindings, curr) => {
-      const acc = prev;
-      const level = curr!.weaknessRiskLevel as ThreatLevel;
-      // Ignore nulls for non-prod environments, but should always exist on prod
-      if (level === null) return acc;
-      if (!(level in acc)) {
-        acc[level] = 0;
-      }
-      acc[level]! += 1;
-      return acc;
-    },
-    { total: cedarThreat.length }
-  );
+function getSecurityFindings(
+  // eslint-disable-next-line camelcase
+  cedarThreat: GetSystemProfile_cedarThreat[]
+): SecurityFindings {
+  // Init finding props with 0 count
+  const findings = Object.fromEntries(
+    securityFindingKeysOrdered.map(k => [k, 0])
+  ) as SecurityFindings;
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const threat of cedarThreat) {
```

Comment:
Oh, one more comment: this for loop looks good, it feels more understandable to me (not sure how you or other people perceive it). The type assertions are mildly annoying, but that's a Typescript limitation; see https://github.com/microsoft/TypeScript/issues/49305.

Reply:
Yeah, these lines in particular are why I thought of the partial initially.

Improving legibility from this function got me to do the same with a few more things surrounding it. Happy with how it all ended up.